src/main/jbake/content/blog/2016/03/16/hawkular-metrics-roadmap.adoc

```diff
@@ -0,0 +1,49 @@
+= Hawkular Metrics - Roadmap
+Stefan Negrea
+2016-3-16
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, metrics, release
+
+== About
+
+Hawkular Metrics contributors have been working for the past few weeks on a roadmap for the upcoming year. The goal is to give clarity on the project direction, serve as a planning tool for releases, and show our strong commitment to open source.
+
+The project and community are enjoying excellent growth. A maturing code base, an ever growing set of integrations, and consistent community contributions are ingredients that make this project successful and also an indication of the future.
+
+For those not familiar with Hawkular Metrics, the project is a high performance and high availability storage engine for large volume metric data. The project uses Cassandra as a storage engine because of its flexible data model well suited for time-series data storage and linear scalability with no single point of failure.
+
+
+== Roadmap
+
+. **Cassandra 3.x**
+  * The 3.x release of Cassandra is maturing, making the perfect timing for the project to transition from current 2.2.x line
+  * Expect this transition to happen rather soon since work is already in progress (driver updates, and a schema management tool)
+
+. **Pre-computed aggregates**
+  * Needed to support long term data storage and retrieval for high volume metrics
+  * Single metrics roll-ups are also the foundation for pre-computed multi-metric aggregations, that goal is to work on this subsequent to single metric roll-ups
+
+. **Metric Enhancements**
+  * Histogram metrics are fairly common in other time series databases. The plan is to add histogram metrics as a sub-metric to existing gauge metrics, analogous to what counter-rate metrics are counter metrics. It is common to do the calculations need for the histogram on the client side, but there are a lot of advantages to push the calculations to the server.
+  * Add support for metrics baselines; automatically computed server-side and stored
```

Comment:
Also here make a comment that the concrete ideas for the impl need to be researched